test/e2e/eraser_test.go

```diff
@@ -583,8 +588,172 @@ func TestRemoveImagesFromAllNodes(t *testing.T) {
+			if err != nil {
+				t.Errorf("unable to update node %#v with label {%s: %s}\nerror: %#v", nodeToSkip, skipLabelKey, skipLabelValue, err)
+			}
+
+			return ctx
+		}).
+		Assess("Deployment and labelling the node have succeeded", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			c := cfg.Client().RESTConfig()
+			k8sClient, err := clientgo.NewForConfig(c)
+			if err != nil {
+				t.Error("unable to obtain k8s client from config", err)
+			}
+
+			err = wait.For(func() (bool, error) {
+				nodeList, err := k8sClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: skipLabelKey})
+				if err != nil {
+					return false, err
+				}
+
+				return len(nodeList.Items) == 1, nil
+			}, wait.WithTimeout(time.Minute))
+			if err != nil {
+				t.Errorf("error while waiting for selector%s to be added to node\n%#v", skippedNodeSelector, err)
+			}
+
+			resultDeployment := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: nginx, Namespace: cfg.Namespace()},
+			}
+
+			if err = wait.For(
+				conditions.New(cfg.Client().Resources()).DeploymentConditionMatch(&resultDeployment, appsv1.DeploymentAvailable, corev1.ConditionTrue),
+				wait.WithTimeout(time.Minute*3),
+			); err != nil {
+				t.Error("deployment not found", err)
+			}
+
+			return ctx
+		}).
+		Assess("Node(s) successfully skipped", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			//delete deployment
+			client, err := cfg.NewClient()
+			if err != nil {
+				t.Error("Failed to create new client", err)
+			}
+
+			var pods corev1.PodList
+			err = client.Resources().List(ctx, &pods, func(o *metav1.ListOptions) {
+				o.LabelSelector = labels.SelectorFromSet(labels.Set{"app": nginx}).String()
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			dep := ctx.Value(nginx).(*appsv1.Deployment)
+			if err := client.Resources().Delete(ctx, dep); err != nil {
+				t.Error("Failed to delete the dep", err)
+			}
+
+			clusterNodes := getClusterNodes(t)
+			clusterNodes = deleteStringFromSlice(clusterNodes, skippedNodeName)
+
+			for _, nodeName := range clusterNodes {
+				err := wait.For(containerNotPresentOnNode(nodeName, nginx), wait.WithTimeout(time.Minute*2))
+				if err != nil {
+					// Let's not mark this as an error
+					// We only have this to prevent race conditions with the eraser spinning up
+					t.Logf("error while waiting for deployment deletion: %v", err)
+				}
+			}
+
+			// deploy imageJob config
+			if err := deployEraserConfig(cfg.KubeconfigFile(), "eraser-system", "test-data", "eraser_v1alpha1_imagelist.yaml"); err != nil {
+				t.Error("Failed to deploy image list config", err)
+			}
+
+			ctxT, cancel := context.WithTimeout(ctx, time.Minute)
+			defer cancel()
+
+			// ensure images are removed from all nodes except the one we are skipping. remove the node we are skipping from the list of nodes.
+
+			checkImageRemoved(ctxT, t, clusterNodes, nginx)
+
```

Comment:
Can we check that the image is not removed from the skipped node?

Reply:
Done